### PR TITLE
Refactor CLI Usage Section, add docs

### PR DESCRIPTION
**Pull request by ANeumann82**
More docs for:
- Upgradability
- Immutable Parameters
- Cleanup of CLI Usage section

Signed-off-by: Andreas Neumann <aneumann@mesosphere.com>

### CODE REVIEW

**Comment by meichstedt** [comment]
@ANeumann82 can you please add redirects for the urls that changed (ideally including anchors!) to netlify.toml?

**Comment by ANeumann82** [comment]
> @ANeumann82 can you please add redirects for the urls that changed (ideally including anchors!) to netlify.toml?

Done. I was looking for the correct place to do that :)

File: netlify.toml [hunk 1]
```diff
@@ -17,3 +17,23 @@ to = "/docs/what-is-kudo.html#under-the-hood"
...
+from = "/docs/cli.html#commands"
+to = "/docs/cli/commands.html"
+
+[[redirects]]
+from = "/docs/cli.html#flags"
+to = "/docs/cli/commands.html"
+
+[[redirects]]
+from = "/docs/cli.html#examples"
+to = "/docs/cli/examples.html"
```
Comment (porridge) [inline netlify.toml]: Note that these don't work. `#` is a client-side concept, it does not even get sent to the server.
See for example https://deploy-preview-291--kudo-www.netlify.app/docs/cli.html#commands - still lands at `.../installation.html`

I see two ways around this:
1. leave a stub `content/docs/cli.md` behind, which would contain an 'update your links' note, section names and links to the new places. It's ugly, and not sure if vuepress will even let us have `cli.md` file and `cli` directory together
2. add anchors to `installation.md` somewhere at the bottom, with a single sentence linking to `examples.html` and `commands.html`

Either way these redirects for `#` URIs need to be removed.

File: content/docs/cli/commands.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+Specify the max wait time in seconds for CLI for the update to complete before returning (default "300")
+:::
+
+::: tip Info
+Updating the value of a parameter in an instance will trigger the plan which is specified in the parameter definition.
+:::
+
+## upgrade
+
+Upgrade a KUDO package from the currently installed version to a new version. The upgrade argument must be a name of the
```
Comment (porridge) [inline content/docs/cli/commands.md]:
```suggestion
Upgrade a KUDO operator instance from the currently installed version to a new version. The upgrade argument must be a name of the 
```

File: content/docs/cli/commands.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+The parameter name and value separated by '='. For example `-p NODE_COUNT=3`
+:::
+
+::: flag -P, --parameter-file string
+A YAML file with parameters. See [install](#install) for details.
+:::
+
+::: warning
+Be careful with local directories. If you call `kubectl kudo upgrade flink --instance flink-instance` and you have a local
+directory "flink", the local directory takes precedence over the remote directory.
```
Comment (porridge) [inline content/docs/cli/commands.md]: I don't think this is true any more.

**Review by porridge — CHANGES_REQUESTED** [review]

File: netlify.toml (hunk 1 above)
Comment (ANeumann82) [reply netlify.toml]: I've added a `cli/cli.md` that contains the old anchors and links to the correct subdocuments and added redirect from `cli.html` to `cli/cli.html`, hope this works.

File: content/docs/cli/commands.md (hunk 3 above)
Comment (ANeumann82) [reply content/docs/cli/commands.md]: Good catch, it's not. Fixed it.

File: content/docs/cli/cli.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+How to install and setup KUDO. Details can be found [here](installation.md)
+
+## Commands
+
+An overview about all the things the KUDO CLI can do is [here](commands.md)
+
+## Examples
+
+Easy to follow examples for common use cases are [here](examples.md)
```
Comment (porridge) [inline content/docs/cli/cli.md]: Using `here` as link text is an anti-pattern, see for example https://www.smashingmagazine.com/2012/06/links-should-never-say-click-here/

```suggestion
How to [install and setup KUDO](installation.md).

## Commands

An overview about all [the things the KUDO CLI can do](commands.md).

## Examples

Easy to follow [examples for common use cases](examples.md).
```

**Review by porridge — APPROVED** [review]
LGTM, modulo the inline comment about `here` links.

File: content/docs/cli/cli.md (hunk 4 above)
Comment (porridge) [reply content/docs/cli/cli.md]: Another important reason that is not covered by that article that meaningful link text helps search engines a lot.

Comment (ANeumann82) [reply content/docs/cli/cli.md]: Good point 👍

**Pull request merged**